src/ctia/task/migrate_es_stores.clj

```diff
@@ -0,0 +1,229 @@
+               wildcard)
+    (es-index/delete!
+     (:conn target-store)
+     wildcard))
+  (log/infof "%s - creating index template: %s"
+             (:type target-store)
+             (:indexname target-store))
+  (log/infof "%s - creating index: %s"
+             (:type target-store)
+             (:indexname target-store))
+
+  (es-index/create-template!
+   (:conn target-store)
+   (:indexname target-store)
+   (:config target-store))
+
+  (es-index/create!
+   (:conn target-store)
+   (:indexname target-store)
+   (:settings target-store)))
+
+(defn migrate-store
+  "migrate a single store"
+  [current-store
+   target-store
+   operations
+   batch-size
+   confirm?]
+  (when confirm?
+    (create-target-store target-store))
+
+  (let [store-size (-> (fetch-batch current-store 1 0)
+                       :paging
+                       :total-hits)]
+    (log/infof "%s - store size: %s records"
+               (:type current-store)
+               store-size))
+
+  (loop [offset 0
+         migrated-count 0]
+    (let [{:keys [data paging]
+           :as batch}
+          (fetch-batch current-store
+                       batch-size
+                       offset)
+          next (:next paging)
+          offset (:offset next)
+          migrated (transduce operations conj data)
+          migrated-count (+ migrated-count
+                            (count migrated))]
+      (when (seq migrated)
+        (when confirm?
+          (store-batch target-store migrated))
+
+        (log/infof "%s - migrated: %s documents"
+                   (:type current-store)
+                   migrated-count))
+      (if next
+        (recur offset migrated-count)
+        (log/infof "%s - finished migrating %s documents"
+                   (:type current-store)
+                   migrated-count)))))
+
+(defn migrate-store-indexes
+  "migrate all es store indexes"
+  [prefix ops batch-size confirm?]
+  (let [current-stores (stores->maps @stores)
+        target-stores
+        (source-store-maps->target-store-maps current-stores
+                                              prefix)
+        operations (compose-operations ops)
+        batch-size (or batch-size default-batch-size)]
+
+    (log/infof "migrating stores: %s" (keys current-stores))
+    (log/infof "set batch size: %s" batch-size)
+
+    (doseq [[sk sr] current-stores]
+      (log/infof "migrating store: %s" sk)
+      (migrate-store sr
+                     (sk target-stores)
+                     operations
+                     batch-size
+                     confirm?))))
+
+(defn -main
+  "invoke with lein run -m ctia.task.migrate-es-stores <prefix> <operations> <batch-size> <confirm?>"
+  [prefix ops batch-size confirm?]
+
+  (assert prefix "Please provide an indexname suffix for target store creation")
+  (assert ops "Please provide a csv operation list")
+  (assert batch-size "Please specify a batch size")
+  (log/info "migrating all ES Stores")
+  (setup)
```

Comment:
I would parse the "ops" to a list of keyword before providing them directly to `migrate-store-indexes`.